### ISSUE TITLE
Can I drop rspec support?

### ISSUE THREAD

**Issue by zenspider** [seattlerb/hoe#92]
Is this going to affect _anyone_?

@halostatue? maybe?

**Comment by halostatue** [comment]
Put it in a branch and give me a bit of time to test it against my one gem that uses rspec so I can see what it would take to change.

I’ll need ~2weeks to get to this—life plus computer issues are going to be in the way until at least June 3.

I’m not against it, as I’m pretty sure I can work out the changes to support rspec directly or as a plug-in (hoe-rspec?).

**Comment by zenspider** [comment]
I _suspect_ you're just using rspec's tasks directly.

I have added deprecations that just hit master. Nothing is going away yet, but I'm starting to push on better minitest support and that would involve moving stuff around. I just didn't think any of this stuff is actually used so why not remove it?

**Comment by halostatue** [comment]
Probably. Check diff-lcs if you wanted to check it out before I can.

**Comment by zenspider** [comment]
I just released Hoe 3.18 with deprecation warnings in place. Should be easy-peasy to check now.

**Comment by halostatue** [comment]
```DEPRECATED: I want to drop this entirely. Let me know if you use this! from …/gems/hoe-3.18.0/lib/hoe/test.rb:152:in `define_test_tasks'```

This is what I have to add to my `Rakefile` to make it work:

```ruby
require "rspec/core/rake_task"

desc "Run all specifications"
RSpec::Core::RakeTask.new(:spec) do |t|
  rspec_dirs = %w(spec lib)
  t.rspec_opts = []
  t.rspec_opts << "-I#{rspec_dirs.join(":")}" unless rspec_dirs.empty?
end
task :default => :spec
```

**Comment by halostatue** [comment]
I’m good with this change, but it would be worth putting in the changelog for people to find.

**Comment by zenspider** [comment]
It is? https://github.com/seattlerb/hoe/blob/master/History.rdoc

Or define "it"? Maybe I'm misunderstanding what you want.

**Comment by zenspider** [comment]
Better url: https://github.com/seattlerb/hoe/commit/dd02f882986b07bbe569aef49fd55e62a8a302c3#diff-c65ed44cbe1768797ac07d046c79bdc6

**Comment by halostatue** [comment]
I was suggesting that the History include remediation steps for people upgrading, basically what I did above.

**Comment by halostatue** [comment]
diff-lcs is ready for when you drop this. Is there a way to suppress the warning?

**Comment by zenspider** [comment]
@halostatue when I drop this, I'll probably drop the whole thing, so all the deprecations will go with. Minitest::TestTask is simply better than what Hoe provides these days.

Otherwise a `-W0` should probably turn them off since it just uses `warn`.

**Comment by zenspider** [comment]
`Minitest::TestTask` migrated to minitest a while ago. I just did (I think) the last release of Hoe 3.x. Hoe 4 will drop rspec/testunit/arbitrary test task support. Coming "soon".

I'm open to a way of hooking this to make it more pluggable but as it stands, hoe-rspec could exist independently and not bother hooking into any of the `test` hooks (ie `rake spec` doesn't need to tie into anything to work).

**Comment by zenspider** [comment]
I finally released hoe 4.0.0 that drops rspec and test/unit support (and a bunch of stuff)! WOOT!
